PSMT/File.pm

```diff
@@ -854,7 +816,24 @@ sub RegNewFileTime {
         $sth = $dbh->prepare('INSERT INTO docinfo (fileid, fileext, docid, uptime, uname, srcip, description, shahash, version) VALUES (?, ?, ?, from_unixtime(?), ?, ?, ?, ?, ?)');
         $sth->execute($fileid, $ext, $docid, $uptime, $uname, $srcip, $desc, $hash, $ver);
     }
-    $dbh->db_unlock_tables();
+    if ($sth->rows != 1) {
+        $dbh->db_transaction_rollback();
+        if (defined($perr)) {$$perr = 'file_register_failed'; }
+        return undef;
+    }
+    # move real file
+    my $newpath = $self->GetFilePath($fileid);
+    eval {
+        File::Path::mkpath($newpath);
+    };
+    if (($@) ||  (! rename($fsrc, $newpath . $fileid))) {
+        $dbh->db_transaction_rollback();
```

Comment:
need to check the final file position? may need to add a note on configuration that dav and storage need to in the same logical place